BlueDBOnDisk/src/main/java/io/bluedb/disk/segment/Segment.java

```diff
@@ -40,41 +41,104 @@ public boolean contains(BlueKey key) throws BlueDbException {
 		return get(key) != null;
 	}
 
-	public void save(BlueKey key, T value) throws BlueDbException {
-		ArrayList<BlueEntity<T>> entities = null;
-		File file;
-		try (BlueReadLock<Path> readLock = getReadLockFor(key)) {
-			if (readLock != null) {
-				entities = fetch(readLock);
-				file = readLock.getKey().toFile();
-			} else {
-				entities = new ArrayList<>();
-				file = 	getPathFor(key, 1).toFile();
+	public void update(BlueKey newKey, T newValue) throws BlueDbException {
+		long groupingNumber = newKey.getGroupingNumber();
+		modifyChunk(groupingNumber, new Processor<T>() {

```

Comment:
I don't like creating a new anonymous inner class every time for all of these processes. I'd extract the contents into a class or method and use a lambda or method reference. Or at lease create a private `updateProcessor` member that is an anonymous inner class and can be used every time.